src/Compilers/CSharp/Portable/Symbols/NullableAnnotationExtensions.cs

```diff
@@ -47,10 +48,38 @@ public static NullableAnnotation EnsureCompatible(this NullableAnnotation a, Nul
             return (a < b) ? a : b;
         }
 
-        internal static CodeAnalysis.NullableAnnotation ToPublicAnnotation(this NullableAnnotation annotation)
+        internal static CodeAnalysis.NullableAnnotation ToPublicAnnotation(this NullableAnnotation annotation) => annotation switch
         {
-            Debug.Assert((CodeAnalysis.NullableAnnotation)(NullableAnnotation.Oblivious + 1) == CodeAnalysis.NullableAnnotation.Disabled);
-            return (CodeAnalysis.NullableAnnotation)annotation + 1;
+            NullableAnnotation.Annotated => CodeAnalysis.NullableAnnotation.Annotated,
+            NullableAnnotation.NotAnnotated => CodeAnalysis.NullableAnnotation.NotAnnotated,
+            NullableAnnotation.Oblivious => CodeAnalysis.NullableAnnotation.Disabled,
+            _ => throw ExceptionUtilities.UnexpectedValue(annotation)
+        };
+
+        internal static NullabilityInfo ToNullabilityInfo(this CodeAnalysis.NullableAnnotation annotation, TypeSymbol type)
+        {
+            if (annotation == CodeAnalysis.NullableAnnotation.NotApplicable)
+            {
+                return default;
+            }
+
+            NullableAnnotation internalAnnotation = annotation.ToInternalAnnotation();
+            return internalAnnotation.ToNullabilityInfo(type);
+        }
+
+        internal static NullabilityInfo ToNullabilityInfo(this NullableAnnotation annotation, TypeSymbol type)
```

Comment:
>NullableAnnotation [](start = 63, length = 18)

How is this ```NullableAnnotation``` different from ```CodeAnalysis.NullableAnnotation```? Are these different types? Consider explicitly specifying namespace in this case too, the code will be easier to follow.

Reply:
In general, I think we should avoid having types with same names in different namespaces, with exception of VB vs. C#.

---
In reply to: [270184841](https://github.com/dotnet/roslyn/pull/34429#discussion_r270184841) [](ancestors = 270184841)

Reply:
These are public and internal versions of the same state. And we should have good names for both. For most consumers (other than the translation in SemanticModel), there should not be any ambiguity.

---
In reply to: [270186019](https://github.com/dotnet/roslyn/pull/34429#discussion_r270186019) [](ancestors = 270186019,270184841)

Reply:
The public version has a default state that allows the disambiguation of C# 7 expressions and things that don't have the concept of nullability, such as statements. I had originally renamed the internal version of a type like we have `TypeInfo` and `CSharpTypeInfo`, but I received pushback on that and did this instead.

---
In reply to: [270189832](https://github.com/dotnet/roslyn/pull/34429#discussion_r270189832) [](ancestors = 270189832,270186019,270184841)